internal/statemachine/helper.go

```diff
@@ -816,3 +817,106 @@ func mountFromHost(targetDir, mountpoint string) (mountCmd, umountCmd *exec.Cmd)
 	umountCmd = execCommand("umount", filepath.Join(targetDir, mountpoint))
 	return mountCmd, umountCmd
 }
+
+// manualCopyFile copies a file into the chroot
+func manualCopyFile(copyFileInterfaces interface{}, targetDir string, debug bool) error {
+	copyFileSlice := reflect.ValueOf(copyFileInterfaces)
+	for i := 0; i < copyFileSlice.Len(); i++ {
+		copyFile := copyFileSlice.Index(i).Interface().(*CopyFileType)
+
+		// Copy the file into the specified location in the chroot
+		dest := filepath.Join(targetDir, copyFile.Dest)
+		if debug {
+			fmt.Printf("Copying file \"%s\" to \"%s\"", copyFile.Source, dest)
+		}
+		if err := osutilCopySpecialFile(copyFile.Source, dest); err != nil {
+			return fmt.Errorf("Error copying file \"%s\" into chroot: %s",
+				copyFile.Source, err.Error())
+		}
+	}
+	return nil
+}
+
+// manualExecute executes an executable file in the chroot
+func manualExecute(executeInterfaces interface{}, targetDir string, debug bool) error {
+	executeSlice := reflect.ValueOf(executeInterfaces)
+	for i := 0; i < executeSlice.Len(); i++ {
+		execute := executeSlice.Index(i).Interface().(*ExecuteType)
+		executeCmd := execCommand("chroot", targetDir, execute.ExecutePath)
```

Comment:
I would like to reach out to John about this part. Do you know if in their use case, they actually want to an executable/script from the chroot in the chroot or something from the outside?

Reply:
Ah yes this is something I was wondering as well. The copy-files customization happens first so the user could always copy the script into the chroot and then execute it. I'll check with John though.